pkg/network/master/subnets.go

```diff
@@ -210,20 +191,24 @@ func (master *OsdnMaster) watchNodes() {
 
 func (master *OsdnMaster) handleAddOrUpdateNode(obj, _ interface{}, eventType watch.EventType) {
 	node := obj.(*kapi.Node)
-	nodeIP, err := getNodeIP(node)
-	if err != nil {
-		glog.Errorf("Failed to get node IP for node %s, skipping %s event, node: %v", node.Name, eventType, node)
+
+	var nodeIP string
+	if len(node.Status.Addresses) > 0 && node.Status.Addresses[0].Address != "" {
+		nodeIP = node.Status.Addresses[0].Address
```

Comment:
Do we want to check on NodeAddressType here now, to eg prefer NodeInternalAddress?  I'm assuming that netutils.GetNodeIP() would most likely return the private/internal address for the node given it's a DNS lookup and a cloud provider would likely return the private/internal address rather than the public one.

Otherwise LGTM; better to be more explicit in the node config than rely on "magic" in the master to figure out this stuff.

Reply:
It is possible that node address got from DNS lookup can differ from what cloud provider returns. Currently, kubelet is populating first node address as the NodeInternalIP but I agree that we should not depend on the kubelet internals.
Yes, I will make it explicit here.